slack_connection.go

```diff
@@ -0,0 +1,128 @@
+package pagerduty
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/google/go-querystring/query"
+)
+
+// SlackConnectionConfig is the configuration of a Slack connection as per the documentation
+// https://developer.pagerduty.com/api-reference/c2NoOjExMjA5MzMy-slack-connection.
+type SlackConnectionConfig struct {
+	Events     []string `json:"events"`
+	Urgency    *string  `json:"urgency"`
+	Priorities []string `json:"priorities"`
+}
+
+// SlackConnection is an entity that represents a Slack connections as per the
+// documentation https://developer.pagerduty.com/api-reference/c2NoOjExMjA5MzMy-slack-connection.
+type SlackConnection struct {
+	SourceID   string `json:"source_id"`
+	SourceName string `json:"source_name"`
+	SourceType string `json:"source_type"`
+
+	ChannelID        string `json:"channel_id"`
+	ChannelName      string `json:"channel_name"`
+	NotificationType string `json:"notification_type"`
+
+	Config SlackConnectionConfig `json:"config"`
+}
+
+// SlackConnectionObject is an API object returned by getter functions.
+type SlackConnectionObject struct {
+	SlackConnection
+	APIObject
+}
```

Comment:
We intend to no longer use embedding in this project. Please list the fields on the type explicitly. See #318 for more details.

That said, I think we should probably just elide this type entirely and put all of the fields in the `SlackConnection` type.